modules/nextflow/src/main/groovy/nextflow/fusion/FusionClient.groovy

```diff
@@ -0,0 +1,204 @@
+package nextflow.fusion
+
+import com.google.gson.Gson
+import dev.failsafe.Failsafe
+import dev.failsafe.RetryPolicy
+import dev.failsafe.event.EventListener
+import dev.failsafe.event.ExecutionAttemptedEvent
+import dev.failsafe.function.CheckedSupplier
+import groovy.json.JsonOutput
+import groovy.transform.CompileStatic
+import nextflow.Session
+import nextflow.SysEnv
+import nextflow.exception.AbortOperationException
+import nextflow.fusion.exception.BadResponseException
+import nextflow.fusion.exception.UnauthorizedException
+import nextflow.fusion.exchange.LicenseTokenRequest
+import nextflow.fusion.exchange.LicenseTokenResponse
+import nextflow.util.Threads
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.Executors
+import java.util.function.Predicate
+
+/**
+ * Client service for Fusion-related operations
+ *
+ * @author Alberto Miranda <alberto.miranda@seqera.io>
+ */
+@CompileStatic
+class FusionClient {
```

Comment:
I'd prefer implementing this logic as part of [TowerClient](https://github.com/nextflow-io/nextflow/blob/main/plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerClient.groovy#L57-L57) since we said Platform token is going to be a pre-requisite to use fusion

Reply:
This should be doable having the Tower plugin implementing its own extension for `FusionEnv` see here

https://github.com/nextflow-io/nextflow/blob/main/modules/nextflow/src/main/groovy/nextflow/fusion/FusionEnvProvider.groovy#L40-L45

Reply:
Got it, I separated it because I confused `TowerClient` as being part of the Wave plugin, but if it's part of the tower plugin it makes more sense to have it there.